runtime/server/postgres.go

```diff
@@ -0,0 +1,235 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jackc/pgx/v5/pgtype"
+	wire "github.com/jeroenrinzema/psql-wire"
+	"github.com/marcboeker/go-duckdb"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"github.com/shopspring/decimal"
+)
+
+func (s *Server) QueryHandler(ctx context.Context, query string) (wire.PreparedStatements, error) {
+	clientParams := wire.ClientParameters(ctx)
+	instanceID := clientParams[wire.ParamDatabase]
+	var api *runtimev1.API
+	var err error
+
+	if strings.Contains(query, "pg_attribute") || strings.Contains(query, "pg_catalog") || strings.Contains(query, "pg_type") || strings.Contains(query, "pg_namespace") || !strings.Contains(strings.ToUpper(query), "FROM") {
+		api, err = s.runtime.APIForName(ctx, instanceID, "pg-catalog-sql")
+	} else {
+		// todo how to handle normal SQL ?
+		api, err = s.runtime.APIForName(ctx, instanceID, "metrics-sql")
+	}
```

Comment:
I think this can be simplified quite a bit:
1. I don't really see the reason for setting up `pg-catalog-sql` as a resolver. It's a specific set of logic for the Postgres wire compatible endpoint, not something that needs to support exports or being used in custom APIs/charts/alerts/reports/etc. That logic could just go straight in this file (or maybe in `runtime/server/psql` if it's worthy of a dedicated package)?
    - If you want to use a resolver, then it would make sense to have a single `psql` resolver that parses the query and returns a relevant resolver for it, instead of doing the string matching here.
3. You don't need to call the `metrics-sql` API – that's just for mounting metrics SQL as a standalone API. You can just invoke the `metrics_sql` resolver directly (passing `Resolver == "metrics_sql"` in `ResolveOptions`?
4. As for `how to handle normal SQL ?` – one approach is looking at the first line the SQL and supporting a comment annotation. E.g. BigQuery has `#legacySQL` for supporting its legacy syntax. You could parse for a `-- @connector: xxx` comment, and if found, use that connector directly instead of metrics SQL?

Reply:
1. Sounds good. We can avoid resolver and implement `ResolverResult` interface directly.
2. Sounds good.
3. Can be added in later phases ?

Reply:
3. Sure